packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -1647,14 +1659,30 @@ export abstract class Chart extends Observable implements AgChartInstance {
             this.registerListeners(target, options.listeners);
         }
 
+        const oldGrouping = target.seriesGrouping;
         if (seriesGrouping) {
             target.seriesGrouping = { ...target.seriesGrouping, ...(seriesGrouping as SeriesGrouping) };
         }
+
+        // This probably shouldn't be in this part of the flow
+        // However, the chart won't have access to the previous series grouping value to do this
+        if (applyLayers) {
```

Comment:
I feel this definitely shouldn't exist here - but there's no way for a series to communicate to a chart that its grouping has changed. Maybe something like this:-

```ts
interface SeriesDelegate<T> {
    groupingDidChange: (series: Series<T>) => void;
}

class Series<T> {
    delegate: SeriesDelegate?

      @ActionOnSet<Series<TDatum, TLabel>>({
        changeValue: function (newVal, oldVal) {
            this.onSeriesGroupingChange(oldVal, newVal);
            this.delegate?.groupingDidChange(this)
        },
    })
    seriesGrouping?: SeriesGrouping = undefined;
}
```

Then the chart just has to implement `SeriesDelegate` and assign itself as the delegate to each series

Thoughts?

Reply:
From conversations I has with Alan it seems that we are heading towards using a centralized events-hub, which we'd like to use consistently as a mean of communication between modules as it promotes decoupling and provides easy access points to react to in future development.
So I'd go with managing it using events...

Reply:
Let sync on the need for this change, it's quite high risk and impacts correctness of stacking, grouping and Integrated Charts use cases?